### ISSUE TITLE
ButterflyCount: add a filter to the Picture Guide 

### ISSUE THREAD

**Issue by CrisSevilleja** [BiologicalRecordsCentre/ABLE#245]
Include a filter to the Picture Guide for searching by family. Maybe other characteristics can be added to the filter.

@kazlauskis

**Comment by DavidRoy** [comment]
yes, something similar to the European ladybirds app?

**Comment by CrisSevilleja** [comment]
Exactly. At least to include the filter by family. Later we can add more filters based on the shape, color, and so on.

**Comment by CrisSevilleja** [comment]
People asked me if a filter to search by family or gender can be applied to the Picture Guide

**Comment by DavidRoy** [comment]
@CrisSevilleja did you mean 'gender' or did you mean 'Genus?  I don't think we have images tagged by gender?

**Comment by CrisSevilleja** [comment]
@DavidRoy sorry, I wanted to say Genus

**Comment by kazlauskis** [comment]
Yeah, this wold be cool to have the filters in the app. I wonder if we could do it like in the iRecord Butterflies app so you can both use the filters and search in the species guide by name. What do you think?

![Screenshot 2022-05-16 at 14 44 57](https://user-images.githubusercontent.com/7383162/168586084-5dd9fbde-202d-4fca-bc59-0b2f8b90876b.png)


![Screenshot 2022-05-16 at 14 45 02](https://user-images.githubusercontent.com/7383162/168585967-568b5bf1-9dff-4a4f-a778-3abab0b78936.png)

![Screenshot 2022-05-16 at 14 45 16](https://user-images.githubusercontent.com/7383162/168585973-b92347c0-2178-4fcb-bd3b-b48902683c08.png)

**Comment by CrisSevilleja** [comment]
it would be awesome to add this filter to the ButterflyCount app but I think we don't have the whole European butterfly list with all these attributes... I am afraid

**Comment by kazlauskis** [comment]
No problem, I am wondering about the best way to present the filters. The iRB app filter interface in the screenshots would allow to choose a butterfly Family and we could add any new filters later. We could also do the filters as a popup like in the European Ladybirds app if you wanted. That one doesn't quite work for filters with many choices and it wouldn't have  a species search option though.

**Comment by DavidRoy** [comment]
I like the iRB filters and gives us an incentive to extend the guide filters

**Comment by kazlauskis** [comment]
@CrisSevilleja do you have species families data? In the app we currently use this species dataset:
[species.profiles.csv](https://github.com/BiologicalRecordsCentre/ABLE/files/8824845/species.profiles.csv)

**Comment by kazlauskis** [comment]
Never mind the families data, I found it in the original data set 🙂

**Comment by kazlauskis** [comment]
@CrisSevilleja while I am at it, the species are currently sorted by Family as they originally came from the Euro Butterfly Checklist 2018. Would it not be better to sort the list alphabetically?

**Comment by larspett** [comment]
No, unless that is optional for those who want it, otherwise it will be a mess with unrelated and very different species next to each other

3 juni 2022 kl. 08:48 skrev Karolis Kazlauskis ***@***.***>:

﻿

@CrisSevilleja<https://github.com/CrisSevilleja> while I am at it, the species are currently sorted by Family as they originally came from the Euro Butterfly Checklist 2018. Would it not be better to sort the list alphabetically?
